packages/rolldown/src/plugin/index.ts

```diff
@@ -249,9 +178,54 @@ export type SequentialPluginHooks =
   | 'renderChunk'
   | 'transform'
 
-export type AddonHooks = 'banner' | 'footer' | 'intro' | 'outro'
+export type AddonHooks = '' /* 'banner' | 'footer' | 'intro' | 'outro' */
+
+export type OutputPluginHooks =
+  | 'augmentChunkHash'
+  | 'generateBundle'
+  | 'outputOptions'
+  | 'renderChunk'
+  // | 'renderDynamicImport'
+  | 'renderError'
+  | 'renderStart'
+  // | 'resolveFileUrl'
+  // | 'resolveImportMeta'
+  | 'writeBundle'
 
 export type ParallelPluginHooks = Exclude<
   keyof FunctionPluginHooks | AddonHooks,
   FirstPluginHooks | SequentialPluginHooks
 >
+
+export type PluginHooks = {
```

Comment:
I don't think we need these complex type functionality before we stabllize the final plugin form, with considering adding filter in it.

- It doesn't simplify anything but adding more reading complexity
- We need to add more mechanism into the plugin system. I kind of hope we we do this kind of abstraction after stablizing the plugin system/form.